target_chains/cosmwasm/tools/src/chains-manager/cosmwasm.ts

```diff
@@ -0,0 +1,240 @@
+          sender: accAddress,
+          admin: accAddress,
+          codeId: Long.fromNumber(codeId),
+          label,
+          msg: Buffer.from(JSON.stringify(instMsg)),
+          funds: [],
+        },
+      };
+
+    const txResponse = await this.signAndBroadcastMsg(
+      msgInstantiateContractEncodeObject
+    );
+
+    if (txResponse.rawLog === undefined)
+      throw new Error("error parsing raw logs: rawLog undefined");
+
+    const contractAddr: string = extractFromRawLog(
+      txResponse.rawLog,
+      "_contract_address"
+    );
+
+    return {
+      contractAddr,
+      txHash: txResponse.transactionHash,
+    };
+  }
+
+  async executeContract(
+    req: ExecuteContractRequest
+  ): Promise<ExecuteContractResponse> {
+    const { contractAddr, funds, msg } = req;
+    const msgExecuteContact = {
+      sender: await this.getAddress(),
+      contract: contractAddr,
+      msg: Buffer.from(JSON.stringify(msg)),
+      funds,
+    };
+
+    const msgExecuteContractEncodeObject: MsgExecuteContractEncodeObject = {
+      typeUrl: "/cosmwasm.wasm.v1.MsgExecuteContract",
+      value: msgExecuteContact,
+    };
+
+    const txResponse = await this.signAndBroadcastMsg(
+      msgExecuteContractEncodeObject
+    );
+
+    return {
+      txHash: txResponse.transactionHash,
+    };
+  }
+
+  async migrateContract(
+    req: MigrateContractRequest
+  ): Promise<MigrateContractResponse> {
+    const { newCodeId, contractAddr, migrateMsg } = req;
+    const msgMigrateContractEncodeObject: MsgMigrateContractEncodeObject = {
+      typeUrl: "/cosmwasm.wasm.v1.MsgMigrateContract",
+      value: {
+        sender: await this.getAddress(),
+        contract: contractAddr,
+        codeId: Long.fromNumber(newCodeId),
+        msg: Buffer.from(JSON.stringify(migrateMsg)),
+      },
+    };
+
+    const txResponse = await this.signAndBroadcastMsg(
+      msgMigrateContractEncodeObject
+    );
+
+    if (txResponse.rawLog === undefined)
+      throw new Error("error parsing raw logs: rawLog undefined");
+
+    let resultCodeId = parseInt(
+      extractFromRawLog(txResponse.rawLog, "code_id")
+    );
+
+    assert.strictEqual(newCodeId, resultCodeId);
+
+    return {
+      txHash: txResponse.transactionHash,
+    };
+  }
+
+  async updateContractAdmin(
+    req: UpdateContractAdminRequest
+  ): Promise<UpdateContractAdminResponse> {
+    const { newAdmin, contractAddr } = req;
+    const currAdmin = await this.getAddress();
+
+    const msgUpdateAdminEncodeObject: MsgUpdateAdminEncodeObject = {
+      typeUrl: "/cosmwasm.wasm.v1.MsgUpdateAdmin",
+      value: {
+        sender: currAdmin,
+        newAdmin,
+        contract: contractAddr,
+      },
+    };
+
+    const txResponse = await this.signAndBroadcastMsg(
+      msgUpdateAdminEncodeObject
+    );
+
+    return {
+      txHash: txResponse.transactionHash,
+    };
+  }
+}
+
+// enter key of what to extract
+function extractFromRawLog(rawLog: string, key: string): string {
```

Comment:
is the response json? would we be better off parsing it than searching it with a regex?

you don't have to implement that now, but could be worth a TODO or a code comment for future reference.

Reply:
When parsing we would need to know the complete structure of the object which I don't think we need to know. 
For regex parsing we just need to know that particular part how it looks like.